File: backend/src/Korp.Stock.Api/Models/Product.cs
namespace Korp.Stock.Api.Models;

public sealed class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Code { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public int StockQuantity { get; set; }

    public decimal UnitPrice { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/app/core/models/product.ts
export interface Product {
  id: string;
  code: string;
  description: string;
  stockQuantity: number;
  unitPrice: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductRequest {
  code: string;
  description: string;
  stockQuantity: number;
  unitPrice: number;
}